src/components/namestable/use-worker-info.ts

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as _ from "lodash";
+import { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { ShiftCode } from "../../common-models/shift-info.model";
+import { ContractType, WorkersInfoModel, WorkerType } from "../../common-models/worker-info.model";
+import { ApplicationStateModel } from "../../state/models/application-state.model";
+import { WorkerInfoExtendedInterface } from "./worker-edit/worker-edit.models";
+
+interface UseWorkerInfoReturn {
+  workerInfo: WorkerInfo;
+  setWorkerInfo: (workerInfo: WorkerInfo) => void;
+}
+export function useWorkerInfo(workerName: string): UseWorkerInfoReturn {
+  const [workerInfo, setWorkerInfo] = useState<WorkerInfo>(new WorkerInfo());
+  const getWorkerInfo = <T>(
+    state: ApplicationStateModel,
+    key: keyof WorkersInfoModel
+  ): T | undefined =>
+    state.actualState.persistentSchedule.present.employee_info[key]?.[workerName] as T | undefined;
+  const workerTime = useSelector((state: ApplicationStateModel) =>
+    getWorkerInfo<number>(state, "time")
+  );
+  const workerType = useSelector((state: ApplicationStateModel) =>
+    getWorkerInfo<WorkerType>(state, "type")
+  );
+  const workerContractType = useSelector((state: ApplicationStateModel) =>
+    getWorkerInfo<ContractType>(state, "contractType")
+  );
+  const workerShifts = useSelector(
+    (state: ApplicationStateModel) =>
+      state.actualState.persistentSchedule.present.shifts[workerName]
+  );
+  useEffect(() => {
+    const newWorkerInfo = new WorkerInfo(
+      workerName,
+      workerContractType,
+      workerTime,
+      workerType,
+      workerShifts
+    );
+    setWorkerInfo(newWorkerInfo);
+  }, [workerName, workerContractType, workerTime, workerType, workerShifts]);
+  return {
+    workerInfo,
+    setWorkerInfo,
+  };
+}
+
+export class WorkerInfo {
```

Comment:
Klasa która przechowuje informację o pracowniku. 
Szczerze mówiąc myślałem, że będzie w niej więcej logiki, niż aktualnie.
Nie mniej jednak ja bym to zostawił, bo mi się wydaje, że może się przydać. 
Co wy na to?